### DISCUSSION TITLE
Typescript Generics

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#8272] [Q&A] [answered]
Any ideas how to work with generics?
https://codesandbox.io/s/sharp-resonance-z8io92
I get 3 type errors on `reset`, `handleSubmit` and  `register`

**Comment by Moshyfawn** [comment]
Depends on what you're trying to achieve.

The most basic setup would be [this example](https://codesandbox.io/s/elastic-butterfly-hg98bm?file=%2Fsrc%2FApp.tsx)

**Comment by ghost** [comment]
I'm trying to use same form for multiple types. eventually on submit I would spread original value, so I would get full object, like:
```
function onSubmit(form: T) {
   props.onSubmit({...someObj, ...form});
}
```
Your example does not use generic for component

**Comment by ghost** [comment, marked as answer]
OK seems I found something that can be used. If someone will stumble upon this, here is an example:
https://codesandbox.io/s/weathered-voice-kf3f6v?file=/src/App.tsx
Idea is to use the Base type (in this case FormValues) for hook, but use generic (T extends FormValues) for props. Then just spread both initial value (myObject) and value return from submit handler.

**Comment by DiFuks** [comment]
Still have problem :( https://codesandbox.io/s/react-hook-form-control-forked-w39qfq?file=/src/App.tsx
